### PR TITLE
615/json datasource error

### PR DESCRIPTION
**Pull request by pritipath**
Issue Number: (Link to Github Issue or Azure Dev Ops Task/Story)

## Summary
One sentence summary of PR

## Changes

- List of comprehensive changes

## Checklist

- [ ] Local Tests Passing?
- [ ] CICD and Pipeline Tests Passing?
- [ ] Added any new Tests?
- [ ] Documentation Updates Made?
- [ ] Are there any API Changes? If yes, please describe below.
- [ ] This is not a breaking change. If it is, please describe it below.

## Are there API Changes?
If yes, what are the expected API Changes? Please link to an API-Comparison workflow with the API Diff.

## Is this a breaking change?
If yes, what workflow does this break? 

## Anything else?
Other comments, collaborators, etc.
This PR Closes Issue #

### CODE REVIEW

File: src/CarbonAware.DataSources/CarbonAware.DataSources.Json/src/JsonDataSource.cs [hunk 1]
```diff
@@ -88,7 +88,8 @@ private IEnumerable<EmissionsData> FilterByDateRange(IEnumerable<EmissionsData>
     {
         var (newStartTime, newEndTime) = IntervalHelper.ExtendTimeByWindow(startTime, endTime, MinSamplingWindow);
         var windowData = data.Where(ed => ed.TimeBetween(newStartTime, newEndTime));
-        var filteredData = IntervalHelper.FilterByDuration(windowData, startTime, endTime, TimeSpan.FromHours(DURATION));
+        var duration = GetDurationFromEmissionDataPointsOrDefault(windowData, TimeSpan.FromHours(DURATION));
+        var filteredData = IntervalHelper.FilterByDuration(windowData, startTime, endTime, duration);
```
Comment (bderusha) [inline src/CarbonAware.DataSources/CarbonAware.DataSources.Json/src/JsonDataSource.cs]: Any reason to not resolve this by removing `duration` as an argument entirely and letting `FilterByDuration` just use the duration of the emissions data points?

These additional helper methods used by the `WattTimeDataSource` get used in a context where the operator is not in control of the underlying WattTime data, which is not true for the JSON files.  I would prefer that we be more opinionated about the structure of the json data file and simply expect a duration value for all data points.

**Review by bderusha — CHANGES_REQUESTED** [review]

**Review by bderusha — APPROVED** [review]
:shipit: Thanks for the fast turnaround on this 🔥

**Review by JenMadiedo — APPROVED** [review]
🚢  -it

**Review by juzuluag — APPROVED** [review]
👍

**Pull request merged**